### DISCUSSION TITLE
Animation on imodeljs

### DISCUSSION THREAD

**Discussion by namsancow** [iTwin/itwinjs-core#157] [Q&A]
Now, I have used simple animation tool such as Blender.
I can export my animation work such as FBX format.
FBX format can be easily converted as bim format of imodeljs.
However, I can not search the proper functions for this.
Does imodeljs support animation ?

**Comment by pmconne** [comment]
iModel.js supports rigid body transforms applied over time via [RenderSchedule](https://github.com/imodeljs/imodeljs/blob/06ec145d9e5c15cc70e807b067210d144a5552c9/core/common/src/RenderSchedule.ts#L57) scripts.
It currently does not support skeletal animations.

**Reply by namsancow** [reply]
> iModel.js supports rigid body transforms applied over time via [RenderSchedule](https://github.com/imodeljs/imodeljs/blob/06ec145d9e5c15cc70e807b067210d144a5552c9/core/common/src/RenderSchedule.ts#L57) scripts.
> It currently does not support skeletal animations.

Many thanks to kind response.
After receiving your explanation, I searched some example for RenderSchedule, but I can discovered.
Is there some example for this RenderSchedule ?

I guess that this RenderSchedule is already used in iTwin service as followings. 
Road Design - ContextCapture and ProjectWise ContextShare
https://www.youtube.com/watch?v=1L-SomJzzE8
In above the video, many vehicles move in the mode of rigid body transformation.

**Reply by pmconne** [reply]
The animations in that video are rendered using LumenRT (a non-web-based viewer). I think we could add a [sample](https://www.itwinjs.org/sample-showcase/) demonstrating how to use schedule scripts in iModel.js.

**Reply by namsancow** [reply]
> The animations in that video are rendered using LumenRT (a non-web-based viewer). I think we could add a [sample](https://www.itwinjs.org/sample-showcase/) demonstrating how to use schedule scripts in iModel.js.

Although 2D simple animation is in samples of iTwin.js, I think 3D rigid body transformation sample could be more usefully.
I expect this example will be included in new version of imodel.js.
